File: examples/octave/octra_ex.m
1;

octra;

hello ();

v = DVector ();
v.append (1.0);
v.append (2.0);
v.append (3.5);
printf ("vector size: %d\n", v.size ());
printf ("vector[0]=%f vector[1]=%f\n", v.__paren__ (0), v.__paren__ (1));

p = DPair (10.0, 25.5);
printf ("pair sum: %f\n", p.first + p.second);

File: examples/r/octra_ex.r
library(octrar)

octrar::hello()

# Can only coerce (can't use methods directly without a wrapper)
v <- octrar::DVector(c(0.0, 0.0, 0.0,
    1.0, 0.0, 0.0,
    1.0, 1.0, 0.0,
    0.0, 1.0, 0.0))

# This works as expected
p <- octrar::DPair(1.1, 2.2)
p$first
p$second

# This works as expected
p <- octrar::IPair(1.1, 2.2) # converts to integer
p$first
p$second


